src/Servers.jl

```diff
@@ -268,140 +181,128 @@ e.g.
-                        @error "Error ($i):  $conn" exception=(e, stacktrace(catch_backtrace()))
-                    finally
-                        connectioncounter[] -= 1
-                        close(conn)
-                        @info "Closed ($i):  $conn"
+"main server loop that accepts new tcp connections and spawns async threads to handle them"
+function listenloop(h, server, tcpisvalid, connection_count, reuse_limit, readtimeout, verbose)
+    count = 1
+    while isopen(server)
+        try
+            io = accept(server)
+            io === nothing && continue
+            if !tcpisvalid(io)
+                verbose && @info "Accept-Reject:  $io"
+                close(io)
+                continue
+            end
+            connection_count[] += 1
+            conn = Connection(io)
+            conn.host, conn.port = server.hostname, server.hostport
+            let io=io, count=count
+                @async try
+                    verbose && @info "Accept ($count):  $conn"
+                    handle_connection(h, conn, reuse_limit, readtimeout)
+                    verbose && @info "Closed ($count):  $conn"
+                catch e
+                    if e isa Base.IOError && e.code == -54
+                        verbose && @warn "connection reset by peer (ECONNRESET)"
+                    else
+                        @error exception=(e, stacktrace(catch_backtrace()))
                     end
-                end
-            catch e
-                if e isa Base.IOError
-                    @warn "$e"
-                    break
-                else
-                    rethrow(e)
+                finally
+                    connection_count[] -= 1
+                    close(io)
+                    verbose && @info "Closed ($count):  $conn"
                 end
             end
-            id += 1
-        end
-    catch e
-        if e isa InterruptException
-            @warn "Interrupted: listen($hostname)"
-        else
-            rethrow(e)
+        catch e
+            if e isa InterruptException
+                @warn "Interrupted: listen($server)"
+                close(server)
+                break
+            else
+                rethrow(e)
+            end
         end
-    finally
-        close(tcpserver)
+        count += 1
     end
-
     return
 end
 
-
 """
-Start a timeout monitor task to close the `Connection` if it is inactive.
-Create a `Transaction` object for each HTTP Request received.
+Connection handler: starts an async readtimeout thread if needed, then creates
+Transactions to be handled as long as the Connection stays open. Only reuse_limit + 1
+# of Transactions will be allowed during the lifetime of the Connection.
 """
-function handle_connection(f::Function, c::Connection;
-                           reuse_limit::Int=nolimit,
-                           readtimeout::Int=0, kw...)
-
+function handle_connection(h, c::Connection, reuse_limit, readtimeout)
```

Comment:
What is gained by changing kw args to positional?

Reply:
This is never called by users anyway, so there's not much difference. It _is_, however, easier to inspect code w/ positional arguments rather than keyword arguments, which was the main reason this changed.

Reply:
I'm not sure if I follow your meaning of "inspect". Are you talking about something to do with  @warn_type et all?

Reply:
Yes